Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockGUI.php

```diff
@@ -296,10 +296,20 @@ public function manageObject() : string
         $top_tb->setLeadingImage(ilUtil::getImagePath('arrow_upright.svg'), $this->lng->txt('actions'));
 
         $button = ilSubmitButton::getInstance();
+        $grouped_items = $this->blockView->getItemGroups();
         if ($this->viewSettings->isSelectedItemsViewActive()) {
             $button->setCaption('remove');
         } else {
             $button->setCaption('pd_unsubscribe_memberships');
+            foreach ($grouped_items as $group) {
+                $items = $group->getItems();
+                $group->setItems([]);
```

Comment:
I suggest to add a `filter` method and pass a corresponding callback instead of this kind of workaround.

Reply:
See main PR